SmartTutor/ContentModel/LearningObjects/ChallengeModel/FulfillmentStrategy/ChallengeFulfillmentStrategy.cs

```diff
@@ -1,12 +1,21 @@
 ﻿using RepositoryCompiler.CodeModel.CaDETModel.CodeItems;
 using System.Collections.Generic;
 using System.ComponentModel.DataAnnotations;
+using System.Linq;
 
 namespace SmartTutor.ContentModel.LearningObjects.ChallengeModel.FulfillmentStrategy
 {
     public abstract class ChallengeFulfillmentStrategy
     {
-        [Key] public int Id { get; set; }
-        public abstract bool CheckChallengeFulfillment(List<CaDETClass> solutionAttempt);
+        [Key] public int Id { get; private set; }
+        public List<ChallengeHint> ChallengeHints { get; internal set; }
+
+        public abstract ChallengeEvaluation CheckChallengeFulfillment(List<CaDETClass> solutionAttempt);
+        public abstract List<ChallengeHint> GetHintsForSolutionAttempt(List<CaDETClass> submittedClasses);
```

Comment:
Remove. The children should decide how they will approach constructing the ChallengeEvaluation - no need to expose such a method as a public API.